File: VB/RichEditSendMailSL.Web/App_Code/MailService.vb
Imports Microsoft.VisualBasic
Imports System.IO
Imports System.Text
Imports System.Net.Mail
Imports System.Net.Mime
Imports System.Collections.Generic
Imports System.ServiceModel.Activation

<AspNetCompatibilityRequirements(RequirementsMode := AspNetCompatibilityRequirementsMode.Required)> _
Public Class MailService
	Implements IMailService
	Public Sub SendMail(ByVal mailHost As String, ByVal [from] As String, ByVal [to] As String, ByVal subject As String, ByVal body As String, ByVal attachments As List(Of AttachementInfo)) Implements IMailService.SendMail
		Dim client As New SmtpClient(mailHost, 25)
		Dim afrom As New MailAddress(From, "RichEditClient", System.Text.Encoding.UTF8)
		Dim ato As New MailAddress([to])
		Dim message As New MailMessage(From, [to])

		message.Subject = subject

		Dim htmlView As AlternateView = AlternateView.CreateAlternateViewFromString(body, Encoding.UTF8, MediaTypeNames.Text.Html)

		For i As Integer = 0 To attachments.Count - 1
			Dim info As AttachementInfo = attachments(i)
			Dim resource As New LinkedResource(New MemoryStream(info.Data), info.MimeType)
			resource.ContentId = info.ContentId
			htmlView.LinkedResources.Add(resource)
		Next i

		message.AlternateViews.Add(htmlView)
		message.IsBodyHtml = True

		' Specify your login/password to log on to the SMTP server, if required
		'client.Credentials = new System.Net.NetworkCredential("login", "password");

		client.Send(message)
	End Sub
End Class

File: CS/RichEditSendMailSL.Web/App_Code/MailService.cs
using System.IO;
using System.Text;
using System.Net.Mail;
using System.Net.Mime;
using System.Collections.Generic;
using System.ServiceModel.Activation;

[AspNetCompatibilityRequirements(RequirementsMode = AspNetCompatibilityRequirementsMode.Required)]
public class MailService : IMailService {
    public void SendMail(string mailHost, string from, string to, string subject, string body, List<AttachementInfo> attachments) {
        SmtpClient client = new SmtpClient(mailHost, 25);
        MailAddress afrom = new MailAddress(from, "RichEditClient", System.Text.Encoding.UTF8);
        MailAddress ato = new MailAddress(to);
        MailMessage message = new MailMessage(from, to);

        message.Subject = subject;

        AlternateView htmlView = AlternateView.CreateAlternateViewFromString(body, Encoding.UTF8, MediaTypeNames.Text.Html);

        for (int i = 0; i < attachments.Count; i++) {
            AttachementInfo info = attachments[i];
            LinkedResource resource = new LinkedResource(new MemoryStream(info.Data), info.MimeType);
            resource.ContentId = info.ContentId;
            htmlView.LinkedResources.Add(resource);
        }

        message.AlternateViews.Add(htmlView);
        message.IsBodyHtml = true;

        // Specify your login/password to log on to the SMTP server, if required
        //client.Credentials = new System.Net.NetworkCredential("login", "password");
        
        client.Send(message);
    }
}